File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string log) {
        return log.substr(log.find(" ") + 1);
    }

    std::string log_level(std::string log) {
        return log.substr(log.find("[") + 1, log.find("]") - log.find("[") - 1);
    }

    std::string reformat(std::string log) {
        return message(log) + " (" + log_level(log) + ")";
    }
}

File: solutions/csharp/log-levels/2/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) => logLine.Split(' ', 2)[1].Trim();

    public static string LogLevel(string logLine) => logLine.Substring(1, logLine.IndexOf(']') - 1).ToLower();

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}
